### ISSUE TITLE
Fields being required even after disabling it

### ISSUE THREAD

**Issue by LuizSantos1** [OpenMage/magento-lts#1547]
### Summary (*)

Hi there, I am having a trouble with a customize checkout plugin to comply with new Google Merchant policy. 
Actually, I got a strike already for that.

Now in Brazil there's a new data protection law (LGPD) just like Europe GPDR. So Google Merchant does not allow us 
to keep required for fields such as TELEPHONE, VAT AND DOB. They must be OPTIONAL.

I have changed those fields from "required" to "optional" by goint to: System/Configuration/Customer/Settings/Name and Address Options

It doesn't matter if I put required, optional or no. When I get to the checkout page, these fields are still being required.
I heard there's a DB hack to bypass "is required" field, but I wouldn't like to do that. 

To make sure everything was set up correctly, I have installed a Customer Attribute extension to check if DOB, VAT and Telephone 
fields were set as NOT REQUIRED and they're all set correctly, so something is not working as it was supposed to. 

This is the link of the checkout extension I am using: https://github.com/elisei/moip-transparente-magento

### Examples (*)

To replicate the issue will be a litle bit tricky as this extension is connected to Wirecard Brazil account, so in order
to install and get it working, the tester must have an account with them. But I will describe how I reproduce the issue, OK?

I simulate an item purchase, hit "proceed to checkout", put the e-mail and password
for creating a new customer account, when I get to the customer fill up form, if I hit next without filling up the 
fields mentioned above, it does not allow to go next, because those fields are required. 

Google Merchant team told me it is OK to keep those fields, but they cannot be set as "required" and
I do not know why even though they are enabled as optional they are still being required. Maybe an issue
with OpenMage or the extension? I do not know 
I am sending a screenshot of the issue: https://photos.app.goo.gl/ECUS7ipRvE1PYxdR7

### Proposed solution

Maybe an override or extension to bypass this issue? 
I have tried to solve this with an extension but I haven't found any. Probably this will be an issue soon with 
customers from Germany as well as this particular country has a similar data collect restriction mentioned here.

**Comment by yurirn** [comment]
@LuizSantos1 
o mais fácil a se fazer é só editar o PHTML da sua extensão, para remover esses obrigatórios, é simples.

the easiest thing to do is just edit your extension's PHTML, to remove these mandatory ones, it's simple.

**Comment by LuizSantos1** [comment]
@yurirn 
Eu já tentei, acredito que essa obrigatoriedade seja em arquivos js
E como não manjo muito de jquery, nem me atrevo muito. 

Digo que desconfio, porque fuçando o arquivo js do plugin, encontrei
vários códigos required e validate, ou seja, alguma trigger em js deve detectar
esse campo como vazio e portanto, inválido. E assim, dispara o erro e não deixa
seguir para o passo seguinte. 

Aqui tem alguns exemplos:
https://github.com/elisei/moip-transparente-magento/blob/master/js/MOIP/onestepcheckout/change-events.js 
https://github.com/elisei/moip-transparente-magento/blob/master/js/MOIP/onestepcheckout/jquery.mask.min.js

Translation:

I already tried, I believe that this requirement is in js files
And since I don’t know much about jquery, I don’t dare much.

I say I suspect, because digging through the plugin's js file, I found
several required and validate codes, that is, some trigger in js must detect
this field as empty and therefore invalid. And so, trigger the error and don't let
proceed to the next step.

Here are some examples:
https://github.com/elisei/moip-transparente-magento/blob/master/js/MOIP/onestepcheckout/change-events.js 
https://github.com/elisei/moip-transparente-magento/blob/master/js/MOIP/onestepcheckout/jquery.mask.min.js
